### PR TITLE
Improve misleading `Unexpected "x" in class body.` GDScript parser error

### PR DESCRIPTION
**Pull request by JackErb**
This parser error was misleading.

Fixes:
1. Now points at correct line
2. Now prints out the raw token source for identifiers (e.g. "variable_name" instead of "Identifier")

====

**With Change:**

image

**Without Change (misleading):**

image

* _Bugsquad edit:_ Fixes #96792.

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -1060,8 +1060,8 @@ void GDScriptParser::parse_class_body(bool p_is_multiline) {
 			default:
 				// Display a completion with identifiers.
 				make_completion_context(COMPLETION_IDENTIFIER, nullptr);
-				push_error(vformat(R"(Unexpected "%s" in class body.)", current.get_name()));
 				advance();
+				push_error(vformat(R"(Unexpected "%s" in class body.)", previous.source));
```
Comment (dalexeev) [inline modules/gdscript/gdscript_parser.cpp:1064]: Using the token source code in the error message is debatable, since tokens can be multiline or unreadable (`LITERAL`, `NEWLINE`, `INDENT`, `DEDENT`). I think we should use something like:

```c++
const String token_name = previous.is_identifier() ? previous.get_identifier() : previous.get_name();
```

Or maybe add a new method to `GDScriptTokenizer::Token`.

File: modules/gdscript/tests/scripts/parser/errors/vcs_conflict_marker.gd
Comment (dalexeev) [inline modules/gdscript/tests/scripts/parser/errors/vcs_conflict_marker.gd:1]: Unrelated change.

**Comment by dalexeev** [comment]
Probably fixes (need testing):

* #79659
* #96792

**Comment by JackErb** [comment]
> Probably fixes (need testing):
> 
> * ~[Unexpected "Indent" in class body when the real problem is different, and in another line. #79659](https://github.com/godotengine/godot/issues/79659)~
> * [Error "Unexpected Identfier in class body" shows wrong line #96792](https://github.com/godotengine/godot/issues/96792)

Verified it fixes 96792

File: modules/gdscript/tests/scripts/parser/errors/vcs_conflict_marker.gd
Comment (JackErb) [reply modules/gdscript/tests/scripts/parser/errors/vcs_conflict_marker.gd:1]: Removed this change and made a new test to verify it's on the proper line.

File: modules/gdscript/gdscript_parser.cpp (hunk 1 above)
Comment (JackErb) [reply modules/gdscript/gdscript_parser.cpp]: Updated

File: modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Fixes github.com/godotengine/godot/issues/96792
+
+var test = true
```
Comment (dalexeev) [inline modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd:3]:
```suggestion
```

Comment (JackErb) [reply modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd:3]: This line is here intentionally to verify the Parser Error is pointing at the correct line.

File: modules/gdscript/gdscript_tokenizer.cpp [hunk 3]
```diff
@@ -164,6 +164,15 @@ const char *GDScriptTokenizer::Token::get_name() const {
 	return token_names[type];
 }
 
+String GDScriptTokenizer::Token::get_debug_name() const {
+	switch (type) {
+		case IDENTIFIER:
+			return source;
```
[the remark this answers is not in the document]
Comment (JackErb) [reply modules/gdscript/gdscript_tokenizer.cpp]: Updated to `Identifier "name"`

File: modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd (hunk 2 above)
Comment (JackErb) [reply modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd]: Ended up reformatting it to achieve the same desired effect

Comment (dalexeev) [reply modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd]: Our test system does not allow tracking multiple parser errors. But even without this, we could output line (and column?) for parser errors. This would be better than relying on implicit assumptions. However, this is beyond the scope of this PR, so I suggest not testing it for now.

File: modules/gdscript/gdscript_tokenizer.cpp [hunk 4]
```diff
@@ -164,6 +164,15 @@ const char *GDScriptTokenizer::Token::get_name() const {
 	return token_names[type];
 }
 
+String GDScriptTokenizer::Token::get_debug_name() const {
+	switch (type) {
+		case IDENTIFIER:
+			return vformat(R"(Identifier "%s")", source);
```
Comment (dalexeev) [inline modules/gdscript/gdscript_tokenizer.cpp:170]:
```suggestion
			return vformat(R"(identifier "%s")", source);
```

File: modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Fixes github.com/godotengine/godot/issues/96792
```
Comment (dalexeev) [inline modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd:1]:
```suggestion
# https://github.com/godotengine/godot/issues/96792
```

or

```suggestion
# GH-96792
```

File: modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+
+func test():
+	pass
+
+error = true
```
Comment (dalexeev) [inline modules/gdscript/tests/scripts/parser/errors/unexpected_token_in_class_body.gd:6]:
```suggestion
var error
error = true

func test():
	pass
```

**Review by dalexeev — APPROVED** [review]
Looks good to me.

**Comment by Repiteo** [comment]
Thanks! Congratulations on your first merged contribution! 🎉
